syntax = "proto3";

import "hash.proto";
import "file.proto";

service FTPService {
  rpc UploadFile(stream UploadFileRequest) returns (UploadFileResponse);
}

message UploadFileRequest {
  string filepath = 1;
  uint64 filesize = 2;
  bytes  chunk = 3;
  HashType hashtype = 4;
}

message UploadFileResponse {
  FileMetaData metadata = 1;
  Hash hash = 2;
}
